src/main/java/org/sonatype/nexus/ci/nxrm/v3/ComponentUploaderNxrm3.groovy

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2016-present Sonatype, Inc. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package org.sonatype.nexus.ci.nxrm.v3
+
+import com.sonatype.nexus.api.exception.RepositoryManagerException
+import com.sonatype.nexus.api.repository.v3.DefaultAsset
+import com.sonatype.nexus.api.repository.v3.RepositoryManagerV3Client
+import com.sonatype.nexus.api.repository.v3.formats.maven.MavenComponentBuilder
+
+import org.sonatype.nexus.ci.config.Nxrm3Configuration
+import org.sonatype.nexus.ci.nxrm.ComponentUploader
+import org.sonatype.nexus.ci.nxrm.MavenCoordinate
+
+import groovy.transform.InheritConstructors
+import hudson.model.Result
+
+import static org.sonatype.nexus.ci.util.RepositoryManagerClientUtil.nexus3Client
+
+@SuppressWarnings(['CatchException', 'AbcMetric', 'MethodSize'])
+@InheritConstructors
+class ComponentUploaderNxrm3
+    extends ComponentUploader
+{
+  @Override
+  void upload(final Map<MavenCoordinate, List<RemoteMavenAsset>> remoteMavenComponents,
+                        String nxrmRepositoryId)
+  {
+    def nxrmClient = getRepositoryManagerClient(nxrmConfiguration)
```

Comment:
Maybe check and cast `nxrmConfiguration` as `Nxrm3Configuration` to eliminate the compiler warning.